.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.5.0
+    hooks:
+    -   id: check-json
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+
+-   repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.6
+    hooks:
+    - id: ruff
+      args: [ --fix ]
```

Comment:
In this PR, it seems that pre-commit is also used on the Github Action side, won't this ` --fix` then make code with fixable ruff-errors pass CI?

Reply:
Good point! There is apparently `--exit-non-zero-on-fix`, will add that!

Reply:
Scratch that, seems to work as expected for the hook:
```
ruff.....................................................................Failed
- hook id: ruff
- files were modified by this hook

Found 1 error (1 fixed, 0 remaining).
```